Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -2706,26 +2694,28 @@ private function getForumObjects()
 
     public function checkUsersViewMode() : void
     {
-        if (!isset($_SESSION['viewmode'])) {
-            $_SESSION['viewmode'] = $this->objProperties->getDefaultView();
+        if (!isset($this->selectedSorting)) {
```

Comment:
You initialize this member variable with `public $selectedSorting = ilForumProperties::VIEW_TREE;` So IMO it is **not** possible, that this variable is **not** set.

Reply:
true ^^